app/javascript/components/mentoring/mentor_inbox.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useReducer } from 'react'
+import { ConversationList } from './mentor_inbox/conversation_list'
+import { TrackFilter } from './mentor_inbox/track_filter'
+
+export function reducer(state, action) {
+  switch (action.type) {
+    case 'page.changed':
+      return { ...state, query: { ...state.query, page: action.payload.page } }
+    case 'track.changed':
+      return {
+        ...state,
+        query: { ...state.query, track: action.payload.track, page: 1 },
+      }
+  }
+}
+
+export function MentorInbox({ tracksRequest, ...props }) {
+  const [conversationsRequest, dispatch] = useReducer(
+    reducer,
+    props.conversationsRequest
+  )
+
+  function setPage(page) {
+    dispatch({ type: 'page.changed', payload: { page: page } })
+  }
+
+  function setTrack(track) {
+    dispatch({ type: 'track.changed', payload: { track: track } })
+  }
```

Comment:
You can consider optimising re-renders here by using `useCallback` (as you didn't want to pass `dispatch` down). It would look like this:

```typescript
const setTrack = useCallback((track) => {
  dispatch({ type: 'track.changed', payload: { track: track } })
}, [dispatch])
```

Not a necessity (ie. optimise for fast renders and not for "not rendering".), but something to consider.

Reply:
Thanks! I'll do this.